### PR TITLE
[JIT] Add more python ir utilities

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73329
* #73270
* __->__ #69871

Differential Revision: [D33515232](https://our.internmc.facebook.com/intern/diff/D33515232)

### CODE REVIEW

File: torch/jit/_ir_utils.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        self.guard = None
+
+    def __enter__(self):
+        self.prev_insert_point = self.g.insertPoint()
+        self.g.setInsertPoint(self.insert_point)
+
+    def __exit__(self, *args):
+        self.g.setInsertPoint(self.prev_insert_point)
+
+def with_insert_point(self, insert_point: Union[torch._C.Node, torch._C.Block]):
```
Comment (eellison) [inline torch/jit/_ir_utils.py:17]: Is `with g.with_insert_point` redundant? maybe `with g.insert_point_guard` or something else ? what do people think

**Review by ZolotukhinM — APPROVED** [review]

File: torch/jit/_ir_utils.py (hunk 1 above)
Comment (ZolotukhinM) [reply torch/jit/_ir_utils.py:17]: I like `insert_point_guard` slightly better

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33515232).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33515232).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33515232).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33515232).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33515232).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33515232).

**Pull request closed without merging**